### ISSUE TITLE
Form action not working for Locales

### ISSUE THREAD

**Issue by Sitetools** [tractorcow-farm/silverstripe-fluent#739]
I have a form that works for the default Locale but not the other Locales.
When the form is submitted, on non default Locale the form doesn't submit to the form action.
The form is called WagonCalculatorForm and the action is calculateWagon. 
On submission of the form I can get all the information in the WagonCalculatorForm  method but the calculateWagon doesn't get hit at all when used on a non default Locale.

Any ideas why this would be, I'm at a bit of a loss. 
No errors in the log, the site is in dev mode.

**Comment by tractorcow** [comment]
is the action on a page, or a custom controller?
In the html, what is the url your <form> element is posting too.

You can sometimes force urls / actions to a specific locale by adding `?l=<thelocale>` to it.

**Comment by Sitetools** [comment]
Hi Damian,

It's a form on a page controller.

The URL for the working NZ site is : /nz/products/feed/forage-wagons/premium-forage-wagons/WagonCalculatorForm/
An example of a Locale not working is: /es/products/feed/forage-wagons/premium-forage-wagons/WagonCalculatorForm/
On submit the page submits back to the form but then it doesn't get passed on to the form action method (only for non default Locales).

`public function WagonCalculatorForm() {
     
    $fields = new FieldList(
       ...
      );
    $validator = new RequiredFields();
    $actions = new FieldList();
    $actions->push(FormAction::Create('calculateWagon')->setTitle('Calculate'));
    $form = new Form($this, 'WagonCalculatorForm', $fields, $actions,$validator);

    $form->setTemplate('WagonCalculatorForm');
    $form->loadDataFrom($_REQUEST);
    return $form;
}

public function calculateWagon($data, $form) {
    I am not reaching here for non default Locales
}`

**Comment by tractorcow** [comment]
Yes, this seems to be a serious problem. Nothing that you are using should have broken at all.

What have you got in your WagonCalculatorForm.ss ? 

Also, what version of fluent are you using?

**Comment by tractorcow** [comment]
(I adjusted your comment slightly for readability)

**Comment by Sitetools** [comment]
I'm using 
        "silverstripe/recipe-cms": "~4.9.0@stable",
        "tractorcow/silverstripe-fluent": "^5.0",

This is WagonCalculatorForm.ss

`<form $AttributesHTML>
    <div class="form-group">
        <label class="calculator-label calculator-label-main">Feed Density (KG/M3) $Fields.fieldByName(Feed).Value</label>
        <div id="calculator-wagon-density"></div>
        <input type="hidden" id="calculator-wagon-density-input" value="$Fields.fieldByName(Feed).Value" name="Feed">
    </div>

    <div class="form-group mb-0">
        <div class="row">
            <div class="col-xl-4 col-lg-5 col-md-6 form-col">
                <label class="calculator-label">Number of Animals</label>
                <input type="number" class="form-control" value="$Fields.fieldByName(Animals).Value" name="Animals" placeholder="0">
            </div>

            <div class="offset-lg-1 col-xl-4 col-lg-5 col-md-6 form-col">
                <label class="calculator-label">Dry Matter per Animals (KG)</label>
                <input type="number" class="form-control" value="$Fields.fieldByName(DryMatter).Value" name="DryMatter" placeholder="0.0">
            </div>
        </div>
    </div>

    <div class="form-group mb-0">
        <div class="row">
            <div class="col-xl-4 col-lg-5 col-md-6 form-col">
                <label class="calculator-label">Dry Matter Content (%)</label>
                <input type="number" class="form-control" value="$Fields.fieldByName(DryRatio).Value" name="DryRatio" placeholder="00">
            </div>

            <div class="offset-lg-1 col-xl-4 col-lg-5 col-md-6 form-col">
                <label class="calculator-label">No. Feed Loads per Day</label>
                <input type="number" class="form-control" value="$Fields.fieldByName(Loads).Value" name="Loads" placeholder="0">
            </div>
        </div>
    </div>

    <% if $Actions %>
    <div class="form-group calculator-actions">
        <% loop $Actions %>
        $Field
        <% end_loop %>
        $Fields.fieldByName(FormResult)
    </div>
    <% end_if %>

    <input hidden type="hidden" id="Form_Form_SecurityID" name="SecurityID" value="$securityID">
</form>
`

**Comment by tractorcow** [comment]
Sorry @Sitetools I haven't got around to investigating this. Feel free to leave any feedback if you end up coming up with the cause on your end.

I was hoping to see something in your code, but I think someone will have to check out a clean project and debug it. :)

**Comment by Sitetools** [comment]
No problem, thanks for letting me know.
